### DISCUSSION TITLE
Will there be a lumen 11.x?

### DISCUSSION THREAD

**Discussion by macropay-solutions** [laravel/framework#50857] [General] [answered]
And if yes then when it will be released?

Also does someone know when lumen will be abandoned or what will be its last version?

**Comment by ahinkle** [comment]
There isn't a need for lumen. You should upgrade to Laravel. 

https://x.com/taylorotwell/status/1441101127496323080?s=46

**Reply by macropay-solutions** [reply]
Thank you.
1. Using octane or queue worker in daemon mode needs code refactoring without static and an increased attention to memory leaks, not to mention other implications like singletons etc. Same is valid for Frankenphp in daemon/cached mode. It is easy to write: "do that" but harder to actually accomplish it. Same is valid for using supervisord for workers coupled with saying that the jobs must be idempotent (supervisord might restart the worker after SIGTERM on a deploy or a scale down before the container is stopped resulting in partial executed jobs)-easier said than done.
2. Is nonsense to use Laravel for a microservice. Like the anti framework voices like to say: "You need a wheel but you get the whole car".
3. Opcache + JIT are good candidates to increase the speed of the project.
4. Laravel 11 has the api routes "disabled" by default and they need to be manually published/enabled.
5. Less code => less attack area.
6. Your answer does not answer the actual questions.

Conclusion:
**There is a need for lumen.** 

If the maintainers of lumen decided they will stop maintaining it, then they should say it so other interested entities cam pick over the job from the last version supported. In that way lumen enthusiasts can continue using it without politics involved...

PS. 

If Laravel becomes a framework based on components in which you load ONLY what you NEED and WANT, then you might say there is no need for lumen.
One example: If the developer likes to use php functions for array or string,  then the facades/helpers for these functions are overhead, and those are coming with Laravel whether we like/use them or not.

PS2.

We embraced Lumen and Laravel for a while now but recently we feel like the community is trying to enforce their decisions into our projects.

**Reply by ahinkle** [reply]
The maintainers suggest otherwise. You may have to eventually fork your own version or upgrade to Laravel. Alternatively, you don't have to use all of Laravel as you can install only the required areas in the illuminate namespace.

**Reply by macropay-solutions** [reply]
@ahinkle That is the issue. No one is confirming if or when it will be abandoned. We have only political statements.

_"Alternatively, you don't have to use all of Laravel as you can install only the required areas in the illuminate namespace."_

This will be the lumen 11 then (if possible).

**Reply by mbabker** [reply]
Admittedly, I've never looked at stuff like Lumen or Laravel Zero, but I get the feeling that at least Laravel Zero might become more popular with Laravel itself becoming much more opinionated toward "everything the framework provides will be enabled and you're encouraged to only add to it, not take away from it" and Lumen users being steered toward the main framework and its add-on products (i.e. Octane).  I have apps where I've stripped out a lot of the core framework service providers, or stripped down the HTTP kernel to my bare minimum needs, and the opinionated application builder API that they suggest using nowadays really comes across as that level of customization being an anti-pattern.  So for the folks who want this level of control over their stacks, it's really starting to feel like building off the `laravel/laravel` skeleton or using anything in the `Illuminate\Foundation` namespace is gett to be out of the question.

**Comment by macropay-solutions** [comment, marked as answer]
Lumen 11 is released https://packagist.org/packages/laravel/lumen-framework
But the documentation is not yet up to date
https://lumen.laravel.com/docs/11.x

php version at least

![image](https://github.com/laravel/framework/assets/153634237/c49593c7-7ec4-4c60-b946-9702274b4238)

Interesting how laravel does not have any illuminate requirements...

aa no:
```
    "replace": {
        "illuminate/auth": "self.version",
        "illuminate/broadcasting": "self.version",
        "illuminate/bus": "self.version",
        "illuminate/cache": "self.version",
        "illuminate/collections": "self.version",
        "illuminate/conditionable": "self.version",
        "illuminate/config": "self.version",
        "illuminate/console": "self.version",
        "illuminate/container": "self.version",
        "illuminate/contracts": "self.version",
        "illuminate/cookie": "self.version",
        "illuminate/database": "self.version",
        "illuminate/encryption": "self.version",
        "illuminate/events": "self.version",
        "illuminate/filesystem": "self.version",
        "illuminate/hashing": "self.version",
        "illuminate/http": "self.version",
        "illuminate/log": "self.version",
        "illuminate/macroable": "self.version",
        "illuminate/mail": "self.version",
        "illuminate/notifications": "self.version",
        "illuminate/pagination": "self.version",
        "illuminate/pipeline": "self.version",
        "illuminate/process": "self.version",
        "illuminate/queue": "self.version",
        "illuminate/redis": "self.version",
        "illuminate/routing": "self.version",
        "illuminate/session": "self.version",
        "illuminate/support": "self.version",
        "illuminate/testing": "self.version",
        "illuminate/translation": "self.version",
        "illuminate/validation": "self.version",
        "illuminate/view": "self.version",
        "spatie/once": "*"
    },
```

=>
![image](https://github.com/laravel/framework/assets/153634237/48060b33-bd23-4e42-a5c7-b788afb172f0)

Laravel not in lumen packages:
"illuminate/conditionable": "self.version",
"illuminate/cookie": "self.version",
"illuminate/mail": "self.version",
"illuminate/notifications": "self.version",
"illuminate/process": "self.version",
"illuminate/redis": "self.version",
"illuminate/routing": "self.version",
"illuminate/session": "self.version",
"spatie/once": "*"

So, lumen 8 vs 11 has 2 new packages only:
![image](https://github.com/laravel/framework/assets/153634237/1bf92e5d-2814-458a-b333-3e263bb8310b)

**Reply by inmanturbo** [reply]
@macropay-solutions we've used laravel zero in micro-services. We've also used illuminate packages independently of laravel. Never tried Lumen.

Check this out:
https://github.com/mattstauffer/Torch

Also [leaf](https://leafphp.dev/) works well with illuminate packages.

**Reply by macropay-solutions** [reply]
@inmanturbo we ended up creating [Maravel Framework](https://github.com/macropay-solutions/maravel-framework) inspired by  Lumen 10.0.4 and Laravel Components 10.48.29.

Update.
It will not be deprecated until 31.12.2027 and it is using Symfony components 6.4 (LTS) and php verison 8.1-> 8.3.

**Reply by macropay-solutions** [reply]
Comparing it with a laravel 10 for our crud suite, on local it is faster:

From ~800 ms  to ~600 ms when the "server" is busy and from ~300 ms to ~200 ms when "server" is not too busy when listing 1000 operations from a local mysql db.

Thanks to the improvement in the Container, even if we resolve the classes that use the Macroable trait from container instead of using **new** (giving more control over the framework to the developer to fix bugs or implement new features via external packages) the response times are faster compared to a full Laravel.

The idea is to stop the overhead that gets into the core and only update in the future for the following reasons:
- PHP version,
-  Symfony components LTS versions
-  And possible missed classes that would need to be resolved instead of instantiated directly,

allowing new features and bug fixes via packages, to maintain a stable core that does not need to be updated yearly.

**Reply by macropay-solutions** [reply]
Example of [bug](https://github.com/laravel/framework/discussions/55785) that can be fixed retroactively by registering the UrlGenerator in container with a child that overwrites the hasCorrectSignature method  to use  

```php
return hash_equals($signature, (string) $request->query->filter('signature', ''));
```

**Comment by marius-ciclistu** [comment]
38 Laravel dev commands now available in Lumen: https://github.com/laravel/lumen-framework/issues/1287
